validatehttp/spec.py

```diff
@@ -149,6 +149,22 @@ def matches(self, resp):
                                 'Response header mismatch: {0}'.format(header),
                                 mismatch=(value, resp_value),
                             )
+                elif key == 'content':
+                    if not resp.ok:
+                        # Response was not OK. Ignoring content check.
+                        continue
```

Comment:
Why is this check needed? Maybe I am missing something, but it seems like this check should fail if the response code is an error, as the content body of 4xx or 5xx responses would still have content and could fail this check.

I could also see us forgetting this and writing checks without a response code check. In this case, a 5xx response could pass as valid with this conditional, correct?

Reply:
Good point. I wasn't 100% sure about how this code worked and I thought it would be required here. Now, reading your arguments, I think it's better to remove this check from here 👍🏼